### PR TITLE
add unittest for projectStatusController

### PR DESCRIPTION
**Pull request by yangcao77**
Signed-off-by: Stephanie <stephanie.cao@ibm.com>

This PR is for #47 
This PR added unit tests for projectStatusController.ts. 
The code coverage has increased from 15.36 to 55.16.
Could not test all functions, since some of them require an app container up and running, such as `pingApplication`, and `pingInTransiteApplication`

In addition, function `getProjectStatus` is removed in this PR, since that function is never called/used by any other modules.

### CODE REVIEW

**Comment by yangcao77** [comment]
@rajivnathan @maysunfaisal Can you help review this PR? Thanks!

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    });
+
+    it("isBuildRequired of a non-exist project", async() => {
+        const actualResult = await projectStatusController.isBuildRequired(projectID);
+        expect(actualResult).to.equal(false);
+    });
+
+
+
+    it("getAppState of a non-exist project", async() => {
```
Comment (maysunfaisal) [inline src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: what is the diff between these next three tests and tests from line number 19?

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        expect(actualResult).to.equal(projectStatusController.BuildState.unknown);
+    });
+
+    it("isBuildRequired of a non-exist project", async() => {
+        const actualResult = await projectStatusController.isBuildRequired(projectID);
+        expect(actualResult).to.equal(false);
+    });
+
+    describe("combinational testing of updateProjectStatus function", () => {
+        before("create workspace settings directory with config file", async () => {
```
Comment (maysunfaisal) [inline src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: lets rename the before desc, i think its not correct?

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                        expect(isBuildInProgress).to.equal(true);
+                    }
+                }
+            });
+        }
+    });
+
+
+    describe("combinational testing of buildRequired function", () => {
+        before("create workspace settings directory with config file", async () => {
```
Comment (maysunfaisal) [inline src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: same as above

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        const actualResult = await projectStatusController.isBuildRequired(projectID);
+        expect(actualResult).to.equal(false);
+    });
+
+    describe("combinational testing of updateProjectStatus function", () => {
+        before("create workspace settings directory with config file", async () => {
+            await projectStatusController.addProject(projectID);
+        });
+
+        after("remove the test directory", async () => {
```
Comment (maysunfaisal) [inline src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: I guess we can also say remove project here?

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        }
+    });
+
+
+    describe("combinational testing of buildRequired function", () => {
+        before("create workspace settings directory with config file", async () => {
+            await projectStatusController.addProject(projectID);
+        });
+
+        after("remove the test directory", async () => {
```
Comment (maysunfaisal) [inline src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: same as above

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts (hunk 1 above)
Comment (yangcao77) [reply src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: Testing different functions...

Comment (yangcao77) [reply src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: My bad, That was a dup. Updated

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts (hunk 2 above)
Comment (yangcao77) [reply src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: Updated

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts (hunk 3 above)
Comment (yangcao77) [reply src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: Updated

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts (hunk 4 above)
Comment (yangcao77) [reply src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: Updated

File: src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts (hunk 5 above)
Comment (yangcao77) [reply src/pfe/file-watcher/server/test/unit-test/tests/projectStatusController.test.ts]: Updated

**Pull request merged**